day_10/task_2/pipe_maze.cpp

```diff
@@ -211,16 +210,22 @@ void AttributedMaze::set_tile_at(const MazePoint& point, const Tile& tile)
 
 bool AttributedMaze::is_in_maze(const MazePoint& point) const
 {
-    return point.first >= 0 and point.first < static_cast<int>(maze_[0].size()) and
-        point.second >= 0 and point.second < static_cast<int>(maze_.size());
+    return point.first >= 0 and point.first < static_cast<int>(maze_.size()) and point.second >= 0 and point.second < static_cast<int>(maze_[0].size());
```

Comment:
[optional] 

This change should be commited separately with title in commit like this:

`FIX(AttributedMaze): Swap the dimensions in is_in_maze`

Reply:
let it leave in one commit